python/python/tests/test_remote_db.py

```diff
@@ -31,16 +39,17 @@ def do_POST(self):
 
 @contextlib.contextmanager
 def mock_lancedb_connection(handler):
+    port = find_open_port()
     with http.server.HTTPServer(
-        ("localhost", 8080), make_mock_http_handler(handler)
+        ("localhost", port), make_mock_http_handler(handler)
     ) as server:
         handle = threading.Thread(target=server.serve_forever)
```

Comment:
It might be _slightly_ more efficient to do:

```
with http.servver.HTTPServer(("localhost", 0), make_mock_http_handler(handler) as server:
    port = server.server_port
```

This way you don't have to bind twice.

Reply:
Ah thanks. That's simpler.